src/storage/types.rs

```diff
@@ -0,0 +1,138 @@
+
+        // Increment
+        pub NextDataObjectTypeID get(next_data_object_type_id) build(|config: &GenesisConfig<T>| config.first_data_object_type_id): T::DataObjectTypeID = T::DataObjectTypeID::sa(DEFAULT_FIRST_DATA_OBJECT_TYPE_ID);
+
+        // Mapping of Data object types
+        pub DataObjectTypeMap get(data_object_type): map T::DataObjectTypeID => Option<DataObjectType<T>>;
+    }
+}
+
+decl_event! {
+    pub enum Event<T> where
+        <T as Trait>::DataObjectTypeID
+    {
+        DataObjectTypeAdded(DataObjectTypeID),
+        DataObjectTypeUpdated(DataObjectTypeID),
+    }
+}
+
+
+
+impl<T: Trait> traits::IsActiveDataObjectType<T> for Module<T>
+{
+    fn is_active_data_object_type(which: &T::DataObjectTypeID) -> bool
+    {
+        match Self::ensure_data_object_type(*which)
+        {
+            Ok(dot) => dot.active,
+            Err(_err) => false
+        }
+    }
+}
+
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin
+    {
+        fn deposit_event<T>() = default;
+
+        pub fn register_data_object_type(origin, data_object_type: DataObjectType<T>)
+        {
+            ensure_root(origin)?;
+            ensure!(data_object_type.id.is_none(), MSG_REQUIRE_NEW_DOT);
+
+            let new_dot_id = Self::next_data_object_type_id();
+            let dot: DataObjectType<T> = DataObjectType {
+                id: Some(new_dot_id),
+                description: data_object_type.description.clone(),
+                active: data_object_type.active,
+            };
+
+            <DataObjectTypeMap<T>>::insert(new_dot_id, dot);
+            <NextDataObjectTypeID<T>>::mutate(|n| { *n += T::DataObjectTypeID::sa(1); });
+
+            Self::deposit_event(RawEvent::DataObjectTypeAdded(new_dot_id));
+        }
+
+        pub fn update_data_object_type(origin, data_object_type: DataObjectType<T>)
+        {
+            ensure_root(origin)?;
+            ensure!(data_object_type.id.is_some(), MSG_REQUIRE_DOT_ID);
+
+            let id = data_object_type.id.unwrap();
+            let mut dot = Self::ensure_data_object_type(id)?;
+
+            dot.description = data_object_type.description.clone();
+            dot.active = data_object_type.active;
+
+            <DataObjectTypeMap<T>>::insert(id, dot);
+
+            Self::deposit_event(RawEvent::DataObjectTypeUpdated(id));
+        }
+
+        pub fn activate_data_object_type(origin, id: T::DataObjectTypeID, active: bool)
+        {
+            ensure_root(origin)?;
+            let mut dot = Self::ensure_data_object_type(id)?;
+
+            dot.active = active;
+
+            <DataObjectTypeMap<T>>::insert(id, dot);
+
+            Self::deposit_event(RawEvent::DataObjectTypeUpdated(id));
+        }
+    }
+}
+
+impl <T: Trait> Module<T>
+{
+    fn ensure_data_object_type(id: T::DataObjectTypeID) -> Result<DataObjectType<T>, &'static str>
+    {
+        let dot = Self::data_object_type(&id).ok_or(MSG_DOT_NOT_FOUND)?;
+        Ok(dot)
```

Comment:
It can be one liner expression, since ok_or() returns a `Result`
```
fn ensure_data_object_type(id: T::DataObjectTypeID) -> Result<DataObjectType<T>, &'static str> {
    Self::data_object_type(&id).ok_or(MSG_DOT_NOT_FOUND)
}
```

Reply:
Thanks! I missed that.

Reply:
Fixed in another commit.